lifecycle.md

```diff
@@ -48,9 +48,44 @@ Note: this content should be moved to another location as it is not lifecycle re
 
 Build and test images are referenced in repo infra files, across a variety of `main` and `release/*` branches. Updating these references is a multi-step detail-oriented task. It is a pain, but necessary.
 
-We use version specific references in infra to ensure that our CI builds are reliable. One can imagine using floating OS tags (such as `debian-oldest` and `debian-latest`), however such an approach would be guaranteed to break our build. We know that since we often see build and test breaks that need addressing in PRs where we update build and test images.
+Two types of tag styles are available: [distro version-specific](#distro-version-specific-tags) and [floating](#floating-tags).
 
-At times, it may be necessary to use a [fixed image reference](https://github.com/dotnet/runtime/pull/110199#discussion_r1859075989) for build reliability. If this is ever done, a tracking issue should be created (before the PR is merged) so that we remember to resolve the underlying issue and update the image reference.
+### Distro Version-Specific Tags
+
+Distro version-specific tags include the distro's version (e.g. `alpine-3.21-helix-amd64`).
+Repos that are susceptible to breaking changes in the distro should use these tags.
+[dotnet/runtime](https://github.com/dotnet/runtime) is an example of such a repo.
+
+### Floating Tags
+
+Floating tags have no distro version indicated in the name and are scoped to a .NET version (e.g. `alpine-net9.0-helix-amd64`).
+It is routinely updated to reference a new version as the distro's and .NET's lifecycles progress.
+
+Floating tags are provided on an as-needed basis.
```

Comment:
We could use some text in this section that we'll stop re-building these tags once the given release goes EOL.